### PR TITLE
reports: Display sha256sum on webpage

### PR DESCRIPTION
**Pull request by KVGarg**
As per issue, I have added the computed sha256sum hash values for
patch tarball for all the 2017 report files and display them on the
project report webpage.

Closes https://github.com/coala/projects/issues/627

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: _layouts/report.md [hunk 1]
```diff
@@ -82,6 +81,12 @@
                             <div class="center"><h5 class="light no-margin">Patches Tarball</h5></div>
                             <br>
                             <div class="center"><a href="{{ page.tarball }}">Download</a></div>
+                            <br>
+                            <div class="center">
+                                <p style="word-wrap: break-word;" >
+                                    <strong>SHA-256</strong>: {{page.sha256sum}}
```
Comment (li-boxuan) [inline _layouts/report.md]: IMO the best design is that you have an abbreviation (say, the first seven letters) displayed, and a button beside it, which people can click to copy the hash. Example: https://github.com/coala/projects/pull/689/commits (see the hash and copy icon)

File: _layouts/report.md [hunk 2]
```diff
@@ -69,7 +69,6 @@
                                 <div class="col m3"><i class="fa fa-envelope-o"></i></div>
                                 <div class="col m9"><a href="mailto:{{ page.email }}">{{ page.email }}</a></div>
                             </div>
-
```
Comment (li-boxuan) [inline _layouts/report.md]: Please remove unnecessary change 😉

File: _layouts/report.md (hunk 1 above)
Comment (li-boxuan) [reply _layouts/report.md]: But this is debatable; feel free to discuss here

Comment (KVGarg) [reply _layouts/report.md]: Ok, I will create this design and will post its screenshot with some `js` included in `report.md` to implement that `copy` feature.

Comment (KVGarg) [reply _layouts/report.md]: image
Is this ok or you want me to implement same as GitHub has done ?

Comment (KVGarg) [reply _layouts/report.md]: @li-boxuan  Also, I have added some Javascript to implement the `copy` feature, so should I create a new file or remain that `js code` in the `html` document only. The existing `JS` files include app related functionality and no `JS` file has been created to handle or manipulate the HTML elements so either we have to create a new file or keep that code in the `html code` in the end of body.

Comment (li-boxuan) [reply _layouts/report.md]: Can you push what you have so far so that reviewers apart from me can comment on it?

Comment (KVGarg) [reply _layouts/report.md]: Ok

Comment (KVGarg) [reply _layouts/report.md]: image

**Comment by KVGarg** [comment]
Done 😄

File: _layouts/report.md [hunk 3]
```diff
@@ -82,6 +82,14 @@
                             <div class="center"><h5 class="light no-margin">Patches Tarball</h5></div>
                             <br>
                             <div class="center"><a href="{{ page.tarball }}">Download</a></div>
+                            <br>
+                            <div class="center sha256sum_hash">
+                                <strong>SHA-256</strong>:
+                                <p id="sha256sum_hash_value">
+                                    {{page.sha256sum}}
+                                </p>
+                                <i class="fa fa-clipboard" aria-hidden="true"></i>
```
Comment (li-boxuan) [inline _layouts/report.md]: Can you add hover effect here - at least the cursor should become a pointer when hovering.

Comment (KVGarg) [reply _layouts/report.md]: Will commit soon with hover effect on button :-)

Comment (KVGarg) [reply _layouts/report.md]: screenshot 40
On hovering over icon, the cursor changes to a pointer and a hint text is being displayed. Looks Good to me

Comment (li-boxuan) [reply _layouts/report.md]: Looks good, can you push your changes 👍

Comment (KVGarg) [reply _layouts/report.md]: Pushed my changes 😃

File: resources/css/style.css [hunk 4]
```diff
@@ -85,3 +85,40 @@
   border: 0;
   z-index: 9;
 }
+.sha256sum_hash {
+    display: flex;
```
Comment (jayvdb) [inline resources/css/style.css]: you have different types of indentation in your patch

Comment (jayvdb) [reply resources/css/style.css]: also alpha order the properties

Comment (KVGarg) [reply resources/css/style.css]: Ok @jayvdb , I will make these changes and commit them soon :-)

Comment (KVGarg) [reply resources/css/style.css]: Some other properties that I haven't added are not in alpha order, should I arrange them too ?

Comment (KVGarg) [reply resources/css/style.css]: I have committed changes with alpha ordering of all the properties and proper indentation. Please check if any more change is needed :-)

**Comment by KVGarg** [comment]
@jayvdb @li-boxuan any more changes need to be done ?

**Comment by KVGarg** [comment]
@jayvdb @li-boxuan Any more changes needed ?

File: _layouts/report.md [hunk 5]
```diff
@@ -147,5 +155,21 @@
             </div>
             <br><br>
         </div>
+        <script>
+        $('document').ready(function(){
+            var hash_value = document.getElementById('sha256sum_hash_value').textContent.trim();
+            document.getElementById('sha256sum_hash_value').textContent = hash_value.substring(0,7);
+            document.querySelector(".fa-clipboard").addEventListener("click", function(){
```
Comment (jayvdb) [inline _layouts/report.md]: you use single quotes and double quotes in JS ... which one should be used?  consistency is important.

File: _layouts/report.md [hunk 6]
```diff
@@ -147,5 +155,21 @@
...
+                var new_elment = document.createElement('textarea');
+                new_elment.value = hash_value;
+                new_elment.setAttribute('readonly', '');
+                new_elment.style = {position: 'absolute', left: '-9999px'};
```
Comment (jayvdb) [inline _layouts/report.md]: style should be in CSS

File: _layouts/report.md (hunk 5 above)
Comment (KVGarg) [reply _layouts/report.md]: Since there is a choice to programmer which quoting style he prefers in JavaScript code. To maintain the consistency over all `.js` files I will make changes to single quotes 😃

File: _layouts/report.md (hunk 6 above)
Comment (KVGarg) [reply _layouts/report.md]: The element is created and then deleted, it is just used to copy the hash value once it's get copied the element is deleted from the body.

**Comment by KVGarg** [comment]
Done 👍

File: _layouts/report.md [hunk 7]
```diff
@@ -147,5 +155,21 @@
             </div>
             <br><br>
         </div>
+        <script>
+        $('document').ready(function(){
+            var hash_value = document.getElementById('sha256sum_hash_value').textContent.trim();
+            document.getElementById('sha256sum_hash_value').textContent = hash_value.substring(0,7);
+            document.querySelector('.fa-clipboard').addEventListener('click', function(){
+                var new_element = document.createElement('textarea');
+                $("new_element:first").addClass("hash_value_dup");
```
[the remark this answers is not in the document]
Comment (KVGarg) [reply _layouts/report.md]: Sorry, will commit this change 😄

[the remark this answers is not in the document]
Comment (KVGarg) [reply _layouts/report.md]: Committed 👍

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/689/commits/242912e77a5750c65c6eba60e37177e25faeae91

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
squash your commits.  we dont use `"` in one commit and then use `'` in the next.

And this is all one feature - no need to split into two commits.

**Comment by KVGarg** [comment]
I have squashed the commits, but I haven't understand what do you mean by  "we don't use `"` in one commit and then use `'` in the next".

**Comment by jayvdb** [comment]
> "we don't use " in one commit and then use ' in the next".

your first commit used `"` and then you 'fixed' it in the second commit, using `'` instead.

in coala, we dont do that.  each commit should be complete and correct - atomic.

**Comment by KVGarg** [comment]
Ohh, Ok I got it 👍

**Comment by KVGarg** [comment]
Will change the commit message

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/689/commits/78553b0ab4271e1647fc7f9491e9c66a554d9b2c

**Comment by jayvdb** [comment]
@gitmate-bot ff
